src/main/java/com/epam/test_generator/controllers/CaseController.java

```diff
@@ -172,4 +165,35 @@
 
         return new ResponseEntity<>(HttpStatus.OK);
     }
+
+    @ApiOperation(value = "Perform status changing event on given case", nickname = "performEvent")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "OK"),
+        @ApiResponse(code = 400, message = "Can't perform given event on current case")
+    })
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "suitId", value = "ID of suit which contains the case",
+            required = true, dataType = "long", paramType = "path"),
+        @ApiImplicitParam(name = "caseId", value = "IDs of cases to perform event on",
+            required = true, dataType = "long", paramType = "path"),
+        @ApiImplicitParam(name = "event", value = "Name of event to be performed",
+            required = true, dataType = "String", paramType = "path")
+    })
+    @RequestMapping(value = "suits/{suitId}/cases/{caseId}/events/{event}", method = RequestMethod.PUT,
+        consumes = "application/json")
+    public ResponseEntity<Void> performEvent(@PathVariable("suitId") long suitId,
+                                             @PathVariable("caseId") long caseId,
+                                             @PathVariable("event") String eventName)
+        throws Exception {
+
+        try {
+            Events.valueOf(eventName);
+        } catch (IllegalArgumentException e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+        caseService.performEvent(suitId, caseId, Events.valueOf(eventName));
+
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
```

Comment:
Need controller that returns possible status changes to allow client application display correct list of buttons

Reply:
Added status of case after performing event to controller's response